core/src/test/java/com/rudderstack/android/sdk/core/RudderDataResidencyManagerTest.java

```diff
@@ -0,0 +1,384 @@
+package com.rudderstack.android.sdk.core;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({RudderDataResidencyManager.class})
+public class RudderDataResidencyManagerTest {
+    private RudderDataResidencyManager rudderDataResidencyManager;
+    private RudderServerConfig serverConfig;
+    private RudderConfig config;
+    private final String usDataPlaneUrl = "https://us-dataplane.com";
+    private final String euDataPlaneUrl = "https://eu-dataplane.com";
+    private Map<String, String> dataResidencyUrls = new HashMap<>();
+    private RudderDataResidencyServer rudderDataResidencyServer;
+
+    @Before
+    public void setup() throws Exception {
+        serverConfig = PowerMockito.mock(RudderServerConfig.class);
+        config = spy(PowerMockito.mock(RudderConfig.class));
```

Comment:
What's the use of this?

Reply:
We don't need spy object now. 
For now, I'll mock object will work: `https://github.com/rudderlabs/rudder-sdk-android/pull/149/commits/24ceb87bacea04bd85a80dfe31e2e40237657bc0`. 
I'll make change.

Actually before [this](https://github.com/rudderlabs/rudder-sdk-android/pull/149/commits/24ceb87bacea04bd85a80dfe31e2e40237657bc0) commit, I was actually updating dataPlaneUrl in the config object itself and to test that I had used spy object.